### PR TITLE
Enrich PR summary comment with run metrics, daydream version, and Codex cached tokens

### LINKED ISSUE #65: Enrich PR summary comment with run metrics, daydream version, and Codex cached tokens
## Summary

Daydream's GitHub PR summary comment currently shows only `Mode: {label}` inside its `<details>ℹ️ Review info</details>` block (`daydream/pr_review.py:737-742`). This issue tracks enriching that comment with run-level context — model, cost, token usage, depth, daydream version — so ML/AI researchers, product engineers, and engineering leadership can judge a review's value at a glance.

Full spec: [`.beagle/concepts/enriched-pr-comment/spec.md`](../blob/feat/enriched-pr-comment/.beagle/concepts/enriched-pr-comment/spec.md) (on branch `feat/enriched-pr-comment`).

## Deliverables

- **Visible rollup** (replaces today's single Mode line): Mode, Model (or `mixed — see breakdown`), Cost, Tokens (with cache-hit %), Steps / tool calls.
- **Collapsed per-phase breakdown** as a markdown table inside `<details><summary>Per-phase breakdown</summary>` with columns: Phase, Model, Steps, Tools, Input, Cached, Output, Cost.
- **Footer:** `<sub>Generated by daydream v{version}</sub>`.
- **Cost synthesis from a hardcoded price table** for `gpt-5.5`, `gpt-5.5-pro`, `gpt-5-codex`, `gpt-5.3-codex`. Repeals project decision **D-16**. Claude-backed runs continue to use the SDK-reported `cost_usd`.
- **Codex `cached_input_tokens` extraction fix** — the field is in the protocol (`codex-rs/protocol/src/protocol.rs:2042`) but `daydream/backends/codex.py:297-325` currently hardcodes it to `None`. Bundling the fix here so cache-hit metrics work for both backends from day one.
- **Aggregation across the run**, including parallel fan-out fork trajectory files in deep mode, by reading ATIF artifacts post-run. Renderer becomes a pure function `(trajectory artifacts, mode_label, daydream_version) → markdown`.
- **Graceful fallbacks:** unknown model → `—` + named footnote; missing/unparseable trajectory → today's single Mode line + `*run details unavailable*`. Comment posting must never fail because of metadata problems.
- **Number formatting:** thousand separators ≥1,000; cost <\$0.01 → `<\$0.01`; cache-hit ratio omitted when input tokens = 0.
- **Tests:** extend `test_build_payload_shape` (`tests/test_pr_review.py`) and add edge-case coverage — unknown model, missing trajectory, mixed-model, single-phase, deep-mode multi-trajectory.

## Open questions (from spec)

- **OQ1** — Cached-input prices for `gpt-5.5-pro`, `gpt-5-codex`, `gpt-5.3-codex`: verify against the [OpenAI pricing page](https://openai.com/api/pricing/) at build time; if unpublished, fall back to input-token price (conservative overcount).
- **OQ2** — GPT-5.5's >272K-context multiplier (2× input, 1.5× output): flat price for v1, or per-call multiplier? Spec proposes flat for v1; revisit if real runs commonly exceed 272K input.
- **OQ3** — Phase-name display labels: confirm whether the orchestrator's existing phase keys (`review`, `parse-feedback`, `fix`, `test-and-heal`, `alt-review`, `per-stack-{stack}`, `merge`) are display-quality or need a label dict.
- **OQ4** — Price-table refresh cadence: pre-release checklist item, scheduled review, or ad-hoc?

## Out of scope (already tracked)

- Wall-clock latency per phase → #58
- Cost-per-finding metric → #59
- \`daydream summarize <traj-dir>\` subcommand → #60
- User-overridable price tables → #61
- Per-stack rollups inside deep-mode breakdown → #62
- Reasoning-token column → #63

## Audiences and what they get

| Field | Leadership | Product eng | ML/AI |
|---|---|---|---|
| Mode | context | context | context |
| Model | "is this a serious model?" | trust signal | core fact |
| **Cost** | **the number** | budget signal | compute spend |
| Tokens + cache hit % | skim | skim | core metric — proves caching works |
| Steps / tool calls | "did it work hard?" | depth signal | trajectory shape |
| Per-phase table | skip | "why is fix expensive?" | full audit |
| Version footer | provenance | provenance | reproducibility |

### PR DESCRIPTION
**Pull request by anderskev**
## Summary

Implements the enriched run-info block in daydream's GitHub PR
summary comment per the spec at
`.beagle/concepts/enriched-pr-comment/spec.md`.

Replaces the single-line "Mode" block with:
- Visible rollup: Mode, Model, Cost, Tokens (with cache-hit %),
  Steps / tool calls.
- Collapsed per-phase table: Phase, Model, Steps, Tools, Input,
  Cached, Output, Cost.
- Footer: `Generated by daydream v{version}`.

Bundles the Codex `cached_input_tokens` extraction fix and adds a
hardcoded OpenAI price table (gpt-5.5, gpt-5.5-pro, gpt-5-codex,
gpt-5.3-codex), repealing project decision D-16.

## Test plan

- [ ] Verify `make check` passes
- [ ] Verify rendered comment on a real test PR (manual)
- [ ] Verify Codex run shows non-zero cached tokens
- [ ] Verify deep-mode run aggregates across fork trajectory files
- [ ] Verify unknown-model fallback renders `—` + footnote

Closes #65

### CODE REVIEW

File: daydream/pr_comment_renderer.py [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+def _render_phase_table(agg: _RunAgg) -> list[str]:
+    """Per-phase breakdown inside a collapsed `<details>` block (M2)."""
+    rows: list[str] = [
+        "<details><summary>Per-phase breakdown</summary>",
+        "",
+        "| Phase | Model | Steps | Tools | Input | Cached | Output | Cost |",
+        "|---|---|---|---|---|---|---|---|",
+    ]
+    # Order by first-seen step id so the table reads in execution order.
+    ordered = sorted(agg.phases.values(), key=lambda p: p.first_seen_step_id)
```
[the remark this answers is not in the document]
Comment (anderskev) [reply daydream/pr_comment_renderer.py]: @coderabbitai Fixed in commit 1f0fcbd. Reproduced on the deep-mode fixture: the per-phase table rendered `Review → Fix → Parse Feedback` instead of the actual traversal order `Review → Parse Feedback → Fix` because each fork trajectory restarts `step_id` at 1, so the parent's `Parse Feedback` (step_id=4) ranked behind a fork's `Fix` (step_id=2). Switched to `list(agg.phases.values())` to preserve dict insertion order and dropped the now-unused `first_seen_step_id` field. Regression test at `tests/test_pr_comment_renderer.py:627`.

File: daydream/pr_comment_renderer.py [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+    out = agg.total_output
+    if inp <= 0:
+        return f"{_format_int(inp)} in → {_format_int(out)} out"
+    pct = _format_cache_hit_pct(inp, cached)
+    if cached > 0 and pct is not None:
+        return (
+            f"{_format_int(inp)} in ({_format_int(cached)} cached, {pct} hit) "
+            f"→ {_format_int(out)} out"
+        )
+    return f"{_format_int(inp)} in → {_format_int(out)} out"
```
[the remark this answers is not in the document]
Comment (anderskev) [reply daydream/pr_comment_renderer.py:352]: @coderabbitai Keeping current behavior. Spec M10 only requires omission when input tokens = 0 and is silent on the cached==0 case. Rendering `(0 cached, 0% hit)` when no caching happened adds noise without signal — the call-site docstring at `pr_comment_renderer.py:340-344` documents this as an explicit design decision, and the regression tests at `tests/test_pr_comment_renderer.py:336-338` and `:683-708` lock it in. Both interpretations are spec-defensible; the stricter rule matches the user-facing intent of the rollup.

File: daydream/pr_comment_renderer.py [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+        all_m: set[str] = set()
+        for p in self.phases.values():
+            all_m.update(p.models)
+        return all_m
+
+
+def render_run_info_block(
+    trajectory_paths: list[Path],
+    mode_label: str,
+) -> str:
```
[the remark this answers is not in the document]
Comment (anderskev) [reply daydream/pr_comment_renderer.py:132]: @coderabbitai Keeping current architecture — the version footer IS rendered, just at the comment level rather than inside the renderer. `pr_review.py:817-824` assembles the `<details>ℹ️ Review info</details>` block (renderer output) and `DAYDREAM_FOOTER` (`<sub>🧙 Posted by [daydream v{version}](url)</sub>`, defined at `pr_review.py:151`) as sibling chunks of the comment body. Spec K6 explicitly rejects placing the version "inside the collapsed block (hidden when readers want provenance)" and chooses comment-level footer placement. Adding `daydream_version` to `render_run_info_block` would put the version inside the `<details>` block, directly violating K6.

File: daydream/pr_comment_renderer.py [hunk 4]
```diff
@@ -0,0 +1,439 @@
...
+                continue
+            phase_key = _phase_key_of(step)
+            if phase_key is None:
+                continue
+            phase = _ensure_phase(agg, phase_key, step.step_id)
+            phase.steps += 1
+            phase.tool_calls += len(step.tool_calls or [])
+            if step.model_name:
+                phase.models.add(step.model_name)
+            _accumulate_metrics(agg, phase, step)
```
[the remark this answers is not in the document]
Comment (anderskev) [reply daydream/pr_comment_renderer.py]: @coderabbitai Confirmed — addressed in commit 59f4845. `_aggregate` now reads `agent_default_model = traj.agent.model_name` and threads it through both `effective_model = step.model_name or agent_default_model` (line 210) and `_accumulate_metrics(..., fallback_model=agent_default_model)` (line 213).

File: daydream/pr_comment_renderer.py [hunk 5]
```diff
@@ -0,0 +1,439 @@
...
+        # Step has no model attribution and no SDK-provided cost: cannot
+        # price. Mark unknown so the phase row degrades to '—'.
+        phase.cost_unknown = True
+        return
+    clamped_cached = min(cached, prompt)
+    uncached_input = prompt - clamped_cached
+    synth = compute_cost(
+        model=model,
+        input_tokens=uncached_input,
+        cached_input_tokens=clamped_cached,
```
[the remark this answers is not in the document]
Comment (anderskev) [reply daydream/pr_comment_renderer.py]: @coderabbitai Confirmed — addressed in commit 9bd173c. `_accumulate_metrics` now clamps `prompt`, `completion`, and `cached_raw` to `>= 0`, then applies `cached = min(cached_raw, prompt)` per ATIF v1.6 (cached is a SUBSET of prompt). Regression test at `tests/test_pr_comment_renderer.py:644`.

File: daydream/runner.py [hunk 6]
```diff
@@ -304,7 +304,7 @@ async def run(config: RunConfig | None = None) -> int:
         if codex_in_use:
             quiet = False
     set_quiet_mode(quiet)
-    set_model(config.model or "opus")
+    set_model(config.model or "claude-opus-4-7")
```
[the remark this answers is not in the document]
Comment (anderskev) [reply daydream/runner.py:307]: @coderabbitai Confirmed — addressed in commit 701c1ba. Both `runner.py:307` (`set_model(...)`) and `runner.py:440` (the PR-feedback banner) now default to `claude-opus-4-7`.
